nva-datamodel-java/src/test/java/no/unit/nva/PublicationTest.java

```diff
@@ -117,37 +119,49 @@ void copyReturnsBuilderWithAllDataOfAPublication(Class<?> referenceInstanceType)
         assertThat(diff.prettyPrint(), copy, is(equalTo(publication)));
         assertThat(copy, is(not(sameInstance(publication))));
     }
-    
+
+    @Test
+    void copyShouldCreateDeepCopyOfPublicationWithoutOverridingOriginalPublicationValuesWhenModifyingCopy() {
+        Publication publication = randomPublication();
+        Publication copy = publication.copy().build();
+        EntityDescription entityDescription = publication.getEntityDescription();
+        entityDescription.setDescription(randomString());
+        copy.setLink(randomUri());
+        copy.setEntityDescription(entityDescription);
+        copy.getEntityDescription().getReference().setDoi(randomDoi());
+        assertThat(publication, not(is(equalTo(copy))));
```

Comment:
Possibly use `compareAsObjectNodes` to assert that the actual difference is the expected difference?

Reply:
Also, you will need to bump the model version